Sources/Storage/Disk/Handler/DiskPersistentHandler.swift

```diff
@@ -0,0 +1,120 @@
+//
+//  DiskManager.swift
+//  StatsdClient
+//
+//  Created by Nghia Tran on 10/8/17.
+//  Copyright © 2017 StatsdClient. All rights reserved.
+//
+
+import Foundation
+
+class DiskPersistentHandler: PersistentHandler {
+
+    // MARK: - Variable
+    fileprivate let handler: FileManager
+    fileprivate let path: String
+    fileprivate let defaultEnumerator: [URLResourceKey] = [
+        .isDirectoryKey,
+        .contentModificationDateKey,
+        .totalFileAllocatedSizeKey,
+    ]
+
+    // Config
+    public let config: DiskConfigurable
+
+    // MARK: - Init
+    init(config: DiskConfigurable, hanlder: FileManager = FileManager.default) throws {
+        self.config = config
+        self.handler = hanlder
+
+        // Create path
+        guard let pathFolder = config.pathFolder else {
+            throw DiskError.invalidCacheFolder
+        }
+
+        path = pathFolder
+
+        // Try to create folder
+        try createCacheFolder()
+    }
+
+    // MARK: - Public
+    func makeFilePath(_ key: String) -> String {
+        return "\(path)/\(key.toBase64())"
+    }
+
+    func write<T: Codable>(_ item: T, key: String, attribute: [FileAttributeKey: Any]?) throws {
+        let pathFile = makeFilePath(key)
+        let data = try JSONEncoder().encode(item)
```

Comment:
Move JSONEncoder to a property instead of initializing it every time, also `Decoder`